### PR TITLE
Added capability of specifying delayed_job priority in configuration if used as a backend.

### PR DESCRIPTION
**Pull request by sushantmittal**

### CODE REVIEW

File: lib/devise/async.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ module Backend
     mattr_accessor :enabled
     @@enabled = true
 
+    # Defines the delayed job priority.
+    mattr_accessor :delayed_job_priority
+    @@delayed_job_priority = 0
```
Comment (baschtl) [inline lib/devise/async.rb]: Is it maybe reasonable to use the default priority that is set by Delayed Job itself?
```ruby
Delayed::Worker.default_priority
```

**Comment by sushantmittal** [comment]
By this command, we can set the default priority for all jobs.

But this solution works when we want to set different priority for all devise mails.

**Comment by baschtl** [comment]
@sushantmittal I did not mean to overwrite `Delayed::Worker.default_priority`. What I meant was `@@delayed_job_priority = Delayed::Worker.default_priority` instead of `@@delayed_job_priority = 0`.

**Comment by sushantmittal** [comment]
Oh I see. But it still sets the value 0 because of the following reason:

An application set the delayed_job default_priority in initializer. 
After the above change, default_priority will still get the value 0 that has been set in delayed_job gem instead of the value that has been set in an application.

**Comment by sushantmittal** [comment]
@baschtl 

Can you merge it if looks fine ?

File: README.md [hunk 2]
```diff
@@ -78,6 +78,16 @@ Defaults to :mailer.
 Devise::Async.queue = :my_custom_queue
 ```
 
+### Delayed job priority
+
+Let you specify delayed_job priority if used as a backend.
```
Comment (baschtl) [inline README.md]: `Let's you specify the delayed_job priority if used as a backend.`

File: lib/devise/async/backend/delayed_job.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ module Async
     module Backend
       class DelayedJob < Base
         def self.enqueue(*args)
-          new.delay(:queue => Devise::Async.queue).perform(*args)
+          new.delay(:queue => Devise::Async.queue, priority: Devise::Async.delayed_job_priority).perform(*args)
```
Comment (baschtl) [inline lib/devise/async/backend/delayed_job.rb]: Could you remove the remaining hash rocket here?

**Comment by baschtl** [comment]
Beyond that: +1

**Comment by baschtl** [comment]
@sushantmittal BTW: have a look at https://github.com/mhfs/devise-async/pull/56. IMO it introduces a more general approach to the problem at hand. Hence, we should use this version.

**Comment by sushantmittal** [comment]
@baschtl 

Yeah it looks fine.
I think only one test case missing in ' lib/devise/async.rb' for priority accessor. After that https://github.com/mhfs/devise-async/pull/56 is good to go.

**Comment by baschtl** [comment]
@sushantmittal Ok, #56 is merged. Hence, I will close this PR. Thanks nevertheless for your contribution!

**Pull request closed without merging**